src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1069,6 +1070,123 @@ private void visitKnownTypeValueAlias(String unqualifiedName, String alternative
     typesUsed.add(alternativeAliasName);
   }
 
+  /**
+   * In closure almost any object can be aliased by 'const Something = Other', where other can be
+   * value, type or value-type object. For types (interfaces) and value-type (classes, enums)
+   * TypeScript does not have an easy one-liner syntax. Instead we have to expand into the form:
+   *
+   * <p>type KlassAlias = Klass; const KlassAlias = typeof Klass;
+   *
+   * <p>With incremental clutz it is possible that the type is not present. In such cases one needs
+   * to pass 'null' and an explicit non-null alternativeAliasName will be used.
+   */
+  private void visitTypeValueAlias(String unqualifiedName, ObjectType otype) {
+    String emitName = Constants.INTERNAL_NAMESPACE + "." + otype.getDisplayName();
+    emit("type");
+    emit(unqualifiedName);
+    visitTemplateTypes(otype);
+    emit("=");
+    emit(emitName);
+    visitTemplateTypes(otype, Collections.emptyList(), false);
+    emit(";");
+    emitBreak();
+    if (!otype.isInterface()) {
+      // TS type aliases are only useful in type positions.
+      // To emulate closure alias semantics, introduce also an aliased constructor
+      emit("var " + unqualifiedName);
+      emit(":");
+      emit("typeof");
+      emit(emitName);
+      emit(";");
+      emitBreak();
+    }
+    if (!otype.isInterface() && otype.isFunctionType()) {
```

Comment:
Most of this green block was just copy and pasted to make it visible, but this if statement is new.

Reply:
Ad-hoc pulling some visit* methods out of TreeWalker is not a good idea. Can you introduce a new TreeWalker entry point so that we can to `new TreeWalker(...).specialWalk()`. It would make it easier for me to review this change too.